src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Extensions/ITypeSymbolExtensions.TypeSyntaxGeneratorVisitor.cs

```diff
@@ -300,8 +300,10 @@ public override TypeSyntax VisitNamedType(INamedTypeSymbol symbol)
                     }
                 }
 
-                if (symbol.NullableAnnotation == NullableAnnotation.Annotated)
+                if (symbol is { IsValueType: false, NullableAnnotation: NullableAnnotation.Annotated })
                 {
+                    // value type with nullable annotation may be composed from unconstrained nullable generic
+                    // doesn't mean nullable value type in this case
                     typeSyntax = AddInformationTo(SyntaxFactory.NullableType(typeSyntax), symbol);
                 }
```

Comment:
what about other named types that can be value types.  like tuples.  can you update as well (or add tests showing they're k).

Reply:
Are they special for this case? Here is adding top-level nullability, I couldn't imagine any type to have special semantics. The inner type has been handled above.

Reply:
they're a named type that is a value type :)  i just want to make sure it works properly and nothing unexpected happens.

Reply:
Adding tests should be OK I think